functions/src/plugins/rerun-circleci.ts

```diff
@@ -0,0 +1,69 @@
+import {config} from 'firebase-functions';
+import {Application, Context} from "probot";
+import {Task} from "./task";
+import {RerunCircleCIConfig} from "../default";
+import Github from '@octokit/rest';
+import fetch from "node-fetch";
+
+const CIRCLE_CI_TOKEN = config().circleCI.token;
+
+export class RerunCircleCITask extends Task {
+  constructor(robot: Application, db: FirebaseFirestore.Firestore) {
+    super(robot, db);
+
+    // Dispatch when a label is added to a pull request.
+    this.dispatch([
+      'pull_request.labeled',
+    ], this.checkRerunCircleCI.bind(this));
+  }
+
+  /** Determines if a circle rerun should occur. */
+  async checkRerunCircleCI(context: Context): Promise<void> {
+    if (context.payload.label) {
+      const label: Github.IssuesGetLabelResponse = context.payload.label;
+      const config = await this.getConfig(context);
+      if (config.disabled) {
+        return;
+      }
+      if (label.name === config.triggerRerunLabel) {
+        await this.triggerCircleCIRerun(context);
+      }
+    }
+  }
+
+  /** Triggers a rerun of the default CircleCI workflow and then removed the triggering label. */
+  async triggerCircleCIRerun(context: Context) {
+    const pullRequest: Github.PullRequestsGetResponse = context.payload.pull_request;
+    const config = await this.getConfig(context);
+    if (config.disabled) {
+      return;
+    }
+    const {owner, repo} = context.repo();
+    const circleCiUrl = `https://circleci.com/api/v1.1/project/github/${owner}/${repo}/build?circle-token=${CIRCLE_CI_TOKEN}`;
+    await fetch(circleCiUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        revision: pullRequest.head.sha,
+        branch: `pull/${pullRequest.number}`,
+      })
+    });
```

Comment:
maybe we should catch the request errors and add a comment to the PR that explains that we were unable to restart circle CI because of a request error?